### PR TITLE
SB-25552 - UCI Orchestrator Deployment

### PR DESCRIPTION
**Pull request by shiva-rakshith**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-25552" title="SB-25552" target="_blank">StorySB-25552</a>  UCI - Deployment
  </td></table>
  <br />
 

uci orchestrator deployment configuration

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shiva-rakshith** [comment]
@keshavprasadms Please review the PR.

**Comment by keshavprasadms** [comment]
What will this file contain `ansible/roles/stack-sunbird/templates/orchestrator_rules.xlsx`? I saw the file and we have some data, what is the provider **Gupshup**? How will we handle this file as a code and use ansible templating if we need to have env specific values?

File: kubernetes/ansible/roles/sunbird-deploy/tasks/main.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 - name: create xml configmap
   shell: find "{{role_path}}/templates/" -iname "*.xml" -exec /bin/bash -c "ls {}; kubectl create configmap {{ release_name }}-xml-config --from-file={} -n {{namespace}} --dry-run -o=yaml | kubectl apply -f - " \;
-
+
```
Comment (keshavprasadms) [inline kubernetes/ansible/roles/sunbird-deploy/tasks/main.yml:20]: Please remove whitespaces.

Comment (shiva-rakshith) [reply kubernetes/ansible/roles/sunbird-deploy/tasks/main.yml]: Remove tab spaces.

**Comment by keshavprasadms** [comment]
> What will this file contain `ansible/roles/stack-sunbird/templates/orchestrator_rules.xlsx`? I saw the file and we have some data, what is the provider **Gupshup**? How will we handle this file as a code and use ansible templating if we need to have env specific values?

@shiva-rakshith Without this commit https://github.com/project-sunbird/sunbird-devops/pull/2691/commits/62200940bc538dae32288f258d8132a5532c35da, the xls file is not getting created as a configmap. So your volumeMounts will fail https://github.com/project-sunbird/sunbird-devops/pull/2691/files#diff-5befcb3905191726ce1b555b4e54480bea2123b84845b850b48db5253d273053R52

The question I asked was, if in future we have specific data per environment like dev, stage, production in the xls file, how are we going to achieve that. Since this is a xls file, we wont be able to have ansible / jinja templating.

One option would be to see if we have this data as a csv file and if the application supports. If this option doesn't work with the application, we will need to come up with some solution. Please read on `openpyxl` and `open_excel`. 

At the moment I have no idea what will be data in the xls files, the usecase etc.

**Comment by shiva-rakshith** [comment]
@keshavprasadms I have checked with the team, orchestrator_rules.xlsx need not be an config file. It will be bundled as part of code. I have removed the file and related code changes.

**Review by keshavprasadms — APPROVED** [review]
Looks good

**Pull request merged**